### PR TITLE
feat: show read notifications

### PR DESCRIPTION
**Pull request by setchy**
closes https://github.com/gitify-app/gitify/issues/708

Add read / unread toggle in sidebar.  Read notifications have a higher opacity and hide the mark as read button on hover-over.

Screenshot 2024-03-18 at 8 10 19 AM

Some GitHub API Limitations I've found:
- No endpoint to mark a read notification as `unread`
- No attribute in the `/notifications` response to indicate if a notification is done or not done (ie: in the inbox)

These have been documented in https://github.com/gitify-app/gitify/issues/890

### CODE REVIEW

**Comment by setchy** [comment]
With the improved Settings layout, I've kept this consistent and made this a new Notifications setting

Screenshot 2024-04-03 at 12 45 52 AM

File: src/hooks/useNotifications.test.ts [hunk 1]
```diff
@@ -23,11 +23,11 @@ describe('hooks/useNotifications.ts', () => {
         ];
 
         nock('https://api.github.com')
-          .get('/notifications?participating=false')
+          .get('/notifications?all=false&participating=false')
```
Comment (bmulholland) [inline src/hooks/useNotifications.test.ts:26]: Needing to update the same string throughout is a code smell, FWIW :)

**Review by bmulholland — APPROVED** [review]
These features are getting easy :). Nice stuff!

**Comment by bmulholland** [comment]
If you're happy and you know it, merge conflict

**Comment by setchy** [comment]
Moving back to `DRAFT` as there is some more changes required to ensure that when `showReadNotifications` is enabled that the notification row doesn't disappear, etc

**Comment by setchy** [comment]
Reopening a new PR as I've reworked this

**Pull request closed without merging**